# Part 2: Importing data


### Question 1 - blood transfusion

In [4]:
import pandas as pd
import numpy as np

#1 Import the Pandas library with the alias pd
df= pd.read_csv(r'C:\Users\Gokul\OneDrive\Desktop\UC_Notes\Statistical Computing\lab2_data\blood_transfusion.csv')

#2 What are the dimensions of this data (number of rows and columns)?
df.shape

(748, 5)

In [10]:
#3 What are the data types of each column?
df.dtypes

Recency       int64
Frequency     int64
Monetary      int64
Time          int64
Class        object
dtype: object

In [11]:
#4 Are there any missing values?
df.isna().sum()

Recency      0
Frequency    0
Monetary     0
Time         0
Class        0
dtype: int64

In [13]:
#5 Check out the first 10 rows? What are the Class values for the first 10 observations?
df.head(10)

Recency  Frequency  Monetary  Time        Class
0        2         50     12500    98      donated
1        0         13      3250    28      donated
2        1         16      4000    35      donated
3        2         20      5000    45      donated
4        1         24      6000    77  not donated
5        4          4      1000     4  not donated
6        2          7      1750    14      donated
7        1         12      3000    35  not donated
8        2          9      2250    22      donated
9        5         46     11500    98      donated

#### Class distribution
Donated - 7

NOT Donated - 3

In [14]:
#6 Check out the last 10 rows? What are the Class values for the last 10 observations?
df.tail(10)

Recency  Frequency  Monetary  Time        Class
738       23          1       250    23  not donated
739       23          4      1000    52  not donated
740       23          1       250    23  not donated
741       23          7      1750    88  not donated
742       16          3       750    86  not donated
743       23          2       500    38  not donated
744       21          2       500    52  not donated
745       23          3       750    62  not donated
746       39          1       250    39  not donated
747       72          1       250    72  not donated

#### Class distribution

Donated- 0

Not Donated- 10

In [18]:
#7 Index for the 100th row and just the Monetary column. What is the value?

print('The 100th value of monetary column is ',df['Monetary'][99])

The 100th value of monetary column is  1750


In [20]:
#8 Index for just the Monetary column. What is the mean of this vector?

print('The mean of the Monetary column is ',df['Monetary'].mean())

The mean of the Monetary column is  1378.6764705882354


### Question 2

In [35]:
url = "https://raw.githubusercontent.com/fivethirtyeight/data/master/births/US_births_2000-2014_SSA.csv"
births = pd.read_csv(url)

#1 What are the dimensions of this data (number of rows and columns)?
births.shape

(5479, 5)

In [36]:
#2 Are there any missing values in this data?
births.isna().sum()

year             0
month            0
date_of_month    0
day_of_week      0
births           0
dtype: int64

In [37]:
#3 Index for the 365th row. What is the date of this observation and what was the number of births in the US on this date?

births.iloc[364]

year             2000
month              12
date_of_month      30
day_of_week         6
births           9405
Name: 364, dtype: int64

#### Date of observation is 30th December 2000 and the total number of births for this date was 9405

In [39]:
# Use indexing and the .loc accessor to get the first 31 days in our data frame for just the births column. Is the output a Series or a DataFrame. Now use the .describe() method on this output to compute various summary statistics such as min, max, mean, and median births across the first 31 days in the data

first_31_births = births.loc[:30, 'births']
print(type(first_31_births))

<class 'pandas.core.series.Series'>


#### It is a series

In [40]:
# summary statistics of the first 31 days of the data

first_31_births.describe()

count       31.000000
mean     10894.806452
std       1871.360457
min       7657.000000
25%       8894.500000
50%      11815.000000
75%      12406.500000
max      13032.000000
Name: births, dtype: float64

### Question 3

In [23]:
df2= pd.read_csv(r'C:\Users\Gokul\OneDrive\Desktop\UC_Notes\Statistical Computing\lab2_data\PDI__Police_Data_Initiative__Crime_Incidents.csv')

#1 What are the dimensions of this data (number of rows and columns)?
df2.shape

(15155, 40)

In [27]:
#2 Are there any missing values in this data? If so, how many missing values are in each column? Which column has the most missing values?

#Yes there are missing values in this data.

df2.isna().sum()

INSTANCEID                            0
INCIDENT_NO                           0
DATE_REPORTED                         0
DATE_FROM                             2
DATE_TO                               9
CLSD                                545
UCR                                  10
DST                                   0
BEAT                                 28
OFFENSE                              10
LOCATION                              2
THEFT_CODE                        10167
FLOOR                             14127
SIDE                              14120
OPENING                           14508
HATE_BIAS                             0
DAYOFWEEK                           423
RPT_AREA                            239
CPD_NEIGHBORHOOD                    249
WEAPONS                               5
DATE_OF_CLEARANCE                  2613
HOUR_FROM                             2
HOUR_TO                               9
ADDRESS_X                           148
LONGITUDE_X                        1714


#### OPENING has around 14508 missing values. This is the highest in a column

In [28]:
#3 Index for the DATE_REPORTED column and apply the .describe() method to it. Which date has the most observations in this data set?

df2['DATE_REPORTED'].describe()

count                      15155
unique                     10961
top       04/03/2022 12:59:00 PM
freq                          36
Name: DATE_REPORTED, dtype: object

#### 04/03/2022 has the most observations in this data set

In [30]:
#4 Index for the 100th observation in this data set. What was the date of this crime? What was the offense?

df2.iloc[99]

INSTANCEID                        CDDB7599-DA6E-41F4-A2F8-3B071EC3C5C8
INCIDENT_NO                                                  229000104
DATE_REPORTED                                   01/02/2022 11:23:29 AM
DATE_FROM                                       12/23/2021 11:00:00 AM
DATE_TO                                         12/23/2021 12:00:00 AM
CLSD                                                   Z--EARLY CLOSED
UCR                                                              600.0
DST                                                                  4
BEAT                                                                 6
OFFENSE                                                          THEFT
LOCATION                                                48-PARKING LOT
THEFT_CODE                                  24I-THEFT OF LICENSE PLATE
FLOOR                                                              NaN
SIDE                                                               NaN
OPENIN

In [33]:
df2.iloc[99][['DATE_REPORTED', 'OFFENSE']]

DATE_REPORTED    01/02/2022 11:23:29 AM
OFFENSE                           THEFT
Name: 99, dtype: object

####  The offense was THEFT and the date of the crime was 01.02.2022